app/controllers/mno/extra_mobile_data_requests_controller.rb

```diff
@@ -48,6 +54,24 @@ def bulk_update
     end
   end
 
+  def find_requests
+    @find_requests_form = Mno::FindRequestsForm.new(find_requests_params)
+    if @find_requests_form.phone_number_list.any?
+      redirect_to mno_extra_mobile_data_requests_path(find_requests_params)
+    else
+      @find_requests_form.phone_numbers = nil
+      @find_requests_form.valid?
+      @extra_mobile_data_requests = extra_mobile_data_request_search_scope.order(safe_order)
+      @pagination, @extra_mobile_data_requests = pagy(@extra_mobile_data_requests)
+      @extra_mobile_data_requests_form = Mno::ExtraMobileDataRequestsForm.new(
+        extra_mobile_data_requests: @extra_mobile_data_requests,
+        extra_mobile_data_request_ids: selected_extra_mobile_data_request_ids(@extra_mobile_data_requests, params),
+      )
+      @statuses = mno_status_options
+      render :index
+    end
+  end
```

Comment:
it looks like this has a lot in common with the :index method - if I'm reading this right, it either renders :index or redirects to it - would it make sense to just combine this into the index method, and make the default index page a search with no params?

Reply:
I see what you mean. This really only validates the submitted telephone numbers to check that something valid has been submitted before passing those params off to the index method or rendering an error message. There is a bit of subtlety about how the whole thing fits together though that enables use of the params across the pages.  It felt like it would make the index method a bit unwieldy to do that there (and be submitting form data to the index method that isn't required but you'd like to validate makes it more complex)

Reply:
OK, I'll approve this PR as it's not a blocker - maybe add a Tech Debt card to see if this can be refactored in the firebreak?